### PR TITLE
Fix various issues with using local directories with sources

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: #294

### CODE REVIEW

**Comment by mfojtik** [comment]
@bparees PTAL

**Comment by mfojtik** [comment]
[test]

**Comment by mfojtik** [comment]
Fixes: https://github.com/openshift/source-to-image/issues/294

**Comment by mfojtik** [comment]
(still need tests for all combinations)

**Comment by soltysh** [comment]
I'm ok with changes, but I want to see the tests to get "the tag" from me :wink:

**Comment by mfojtik** [comment]
@soltysh @bparees I'm done with changes and I added the test. Please take a look.

File: pkg/scm/scm.go [hunk 1]
```diff
@@ -13,22 +15,52 @@ import (
...
+	// have GIT binary to fetch it, treat it as file copy.
+	if strings.HasPrefix(s, "file://") && !hasGitBinary() {
+		s = strings.TrimPrefix(s, "file://")
 	}
 
-	glog.Errorf("No downloader defined for %q source URL", s)
-	return nil
+	// If the source is valid GIT protocol (file://, git://, git@, etc..) use GIT
+	// binary to download the sources
+	if g := git.New(); g.ValidCloneSpec(s) {
```
Comment (bparees) [inline pkg/scm/scm.go]: you didn't check if hasGitBinary......

Comment (mfojtik) [inline pkg/scm/scm.go]: if you don't have git binary and it is local (file://) then I strip the "file://" which will make it invalid GIT spec :-)

Comment (mfojtik) [inline pkg/scm/scm.go]: if you specify git:// or git@ protocol and you don't have binary, then you will get an error when attempting to clone.

Comment (bparees) [inline pkg/scm/scm.go]: ok

**Comment by bparees** [comment]
lgtm [merge]

**Pull request merged**